src/wallet/wallet2.cpp

```
@@ -3230,40 +3126,200 @@ static void tree_sync_blocks_async(const TreeSyncStartParams &tree_sync_start_pa
       << " , time spent collecting outs by unlock time while building tree: " << outs_by_last_locked_time_ms_inout / 1000);
 }
 //----------------------------------------------------------------------------------------------------
-void wallet2::process_parsed_blocks(const uint64_t start_height, const std::vector<cryptonote::block_complete_entry> &blocks, const std::vector<parsed_block> &parsed_blocks, uint64_t& blocks_added, std::map<std::pair<uint64_t, uint64_t>, size_t> &output_tracker_cache)
+static bool check_for_reorg(const uint64_t parsed_blocks_start_idx, const crypto::hash &top_hash, const std::vector<tools::wallet2::parsed_block> &parsed_blocks, const hashchain &blockchain, uint64_t &split_point_out, uint64_t &start_parsed_block_i_out)
+{
+  split_point_out = 0;
+  start_parsed_block_i_out = 0;
+
+  const auto set_start_parsed_block_i_out = epee::misc_utils::create_scope_leave_handler([&]() {
+    if (split_point_out == 0/*no reorg*/)
+      return;
+
+    THROW_WALLET_EXCEPTION_IF(!blockchain.is_in_bounds(split_point_out), error::wallet_internal_error,
+      "check_for_reorg: reorg split point not in bounds");
+    THROW_WALLET_EXCEPTION_IF(parsed_blocks_start_idx >= split_point_out, error::wallet_internal_error,
+      "check_for_reorg: reorg split point is expected to be above parsed blocks start");
```

These exceptions will not propagate because they are ran inside `epee::misc_utils::call_befor_die`'s destructor, where all exceptions are ignored (if `epee::misc_utils::call_befor_die` did not do so, it would trigger UB).

---

Nice spot